File: src/DataScience/MATLAB/service_locator.m
function result = service_locator
%SERVICE_LOCATOR Resolve dependencies from a shared registry at runtime.
locator = struct();
locator.email = @(address) "email>" + address;
locator.audit = @(event) "audit>" + event;

email = resolve(locator, "email");
audit = resolve(locator, "audit");
result = struct("email", email("a@example.test"), "audit", audit("created"));
end

function service = resolve(locator, name)
if ~isfield(locator, char(name))
    error("Unknown service: %s", name);
end
service = locator.(char(name));
end

File: src/Enterprise/C#/patterns/ServiceLocator.cs
using System;
using System.Collections.Generic;
namespace Genkidama.PatternExamples;
public static class ServiceLocatorExample { public static bool Run(){var s=new Dictionary<string,Func<string,string>>{{"email",v=>$"email>{v}"},{"audit",v=>$"audit>{v}"}};return s["email"]("a@example.test")=="email>a@example.test"&&s["audit"]("created")=="audit>created";} }
